src/olympia/devhub/tasks.py

```diff
@@ -110,9 +110,9 @@ def create_version_for_upload(addon, upload,
         # review since listed and sideload aren't supported for creation yet.
         if addon.status == amo.STATUS_NULL:
             if disallow_preliminary_review:
-                addon.update(status=amo.STATUS_PUBLIC)
+                addon.update(status=amo.STATUS_NOMINATED)
```

Comment:
> So, hold on. Does that mean that, without this patch, we set add-ons as public without a review somehow ?
> In addition, if that fixes your bug, then why was the issue only happening in the API ?

We do indeed.  But, devhub uploads don't follow this path - the parameter `submit` in `views/handle_upload` is only set to True during an API call, so the sequence of calls that create the version and set the status here don't happen, and the Addon status is set in `views/submit_addon` (for unlisted) or `views/submit_license` (listed) instead.  (Yes, it's totally non-trivial to discover this; yes, it's crazy that we have upload code that is not used in the actual devhub in `devhub/views.py`)